### PR TITLE
[CI environment] Template orchestration for dependencies pipeline - Store VHD to blob container

### PR DESCRIPTION
**Pull request by eriqua**
# Description

- Converting dependencies pipeline jobs to bake and backup a VHD to a storage account from pipeline to template orchestration
- Converting PS jobs to deployment scripts
- Adding image template cleanup
- Updating image template module outputs
- Updating copy to be synchronous
- Updating Convert tokens function to receive an array of files to be converted.

**Note**
- Solving issue blocking PR #1516 (passing strings containing GH secrets between jobs)
- Required updates to pipeline templates and actions will also be leveraged by new token replacement and new dependencies approach

| GH | ADO |
| - | - |
| [.Platform: Dependencies](https://github.com/Azure/ResourceModules/actions/workflows/platform.dependencies.yml) unrelated error with disk encryption set | [Build Status](https://dev.azure.com/carml/CARML/_build/latest?definitionId=105&branchName=users%2Ferikag%2Fdep-deploymentScript) |
| [DesktopVirtualization: ApplicationGroups](https://github.com/Azure/ResourceModules/actions/workflows/ms.desktopvirtualization.applicationgroups.yml) | [Build Status](https://dev.azure.com/carml/CARML/_build/latest?definitionId=30&branchName=users%2Ferikag%2Fdep-deploymentScript) |
# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

# Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (readme)
- [ ] I did format my code

### CODE REVIEW

File: utilities/pipelines/dependencies/constructs/StoreVhdToStorage/deploy.bicep [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  }
+}
+
+// Deploy role assignment
+module roleAssignment '../../../../../modules/Microsoft.Authorization/roleAssignments/subscription/deploy.bicep' = {
+  name: '${uniqueString(deployment().name)}-roleAssignment'
+  params: {
+    roleDefinitionIdOrName: 'Contributor'
+    principalId: userMsi.outputs.principalId
+    subscriptionId: '<<subscriptionId>>'
```
Comment (MrMCake) [inline utilities/pipelines/dependencies/constructs/StoreVhdToStorage/deploy.bicep:30]: Would suggest to use tokens only where we can't avoid it - or in other words, use the capabilities of Bicep/ARM function where we can :)
```suggestion
    subscriptionId: subscription().id
```

File: utilities/pipelines/dependencies/constructs/StoreVhdToStorage/deploy.bicep [hunk 2]
```diff
@@ -0,0 +1,139 @@
+targetScope = 'subscription'
+
+// ================ //
+// Input Parameters //
+// ================ //
+
+// RG parameters
+@description('Optional. The name of the resource group to deploy')
+param resourceGroupName string = 'validation-rg'
```
Comment (MrMCake) [inline utilities/pipelines/dependencies/constructs/StoreVhdToStorage/deploy.bicep:9]: Question: 

Before we leveraged the pipeline variable to set the RG-name. Now it's implemented here and not passed from the outside - which means the user would need to maintain 2 places (which should not be an issue using search & replace).

-> Should we pass the variable as a paramter from the pipeline?

File: utilities/pipelines/dependencies/constructs/StoreVhdToStorage/deploymentScripts/Copy-VhdToStorageAccount.ps1 [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    $imgtRunOutput = Get-AzImageBuilderRunOutput -ImageTemplateName $imageTemplateName -ResourceGroupName $imageTemplateResourceGroup | Where-Object ArtifactUri -NE $null
+    $sourceUri = $imgtRunOutput.ArtifactUri
+    $sourceStorageAccountName = $sourceUri.Split('//')[1].Split('.')[0]
+    $sourceStorageAccount = Get-AzStorageAccount | Where-Object StorageAccountName -EQ $sourceStorageAccountName
+    $sourceStorageAccountContext = $sourceStorageAccount.Context
+    $sourceStorageAccountRGName = $sourceStorageAccount.ResourceGroupName
+    Write-Verbose ('Retrieving artifact uri [{0}] stored in resource group [{1}]' -f $sourceUri, $sourceStorageAccountRGName) -Verbose
+
+    Write-Verbose 'Initializing destination storage account parameters before the blob copy' -Verbose
+    $destinationStorageAccount = Get-AzStorageAccount | Where-Object StorageAccountName -EQ $destinationStorageAccountName
```
Comment (MrMCake) [inline utilities/pipelines/dependencies/constructs/StoreVhdToStorage/deploymentScripts/Copy-VhdToStorageAccount.ps1:93]: Minor thing, but I'd suggest to fetch the storage accounts only once in the script and then filter by name where you need it

File: utilities/pipelines/dependencies/constructs/StoreVhdToStorage/deploymentScripts/Copy-VhdToStorageAccount.ps1 [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    # Copying the VHD to a destination blob container
+    $resourceActionInputObject = @{
+        AbsoluteUri   = $sourceUri
+        Context       = $sourceStorageAccountContext
+        DestContext   = $destinationStorageAccountContext
+        DestBlob      = $destinationBlobName
+        DestContainer = $destinationContainerName
+        Force         = $true
+    }
+    $destBlob = Start-AzStorageBlobCopy @resourceActionInputObject
```
Comment (MrMCake) [inline utilities/pipelines/dependencies/constructs/StoreVhdToStorage/deploymentScripts/Copy-VhdToStorageAccount.ps1:107]: missing the 'ShouldProcess(...)' you correctly specified on top of the `params` block.

File: utilities/pipelines/dependencies/constructs/StoreVhdToStorage/deploymentScripts/Remove-ImageTemplate.ps1 [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        } else {
+            Write-Verbose ('Module [{0}] already installed in version [{1}]' -f $installedModule[0].Name, $installedModule[0].Version) -Verbose
+        }
+    }
+    $VerbosePreference = $currentVerbosePreference
+}
+
+process {
+    # Remove artifacts from existing image template
+    $resourceActionInputObject = @{
```
Comment (MrMCake) [inline utilities/pipelines/dependencies/constructs/StoreVhdToStorage/deploymentScripts/Remove-ImageTemplate.ps1:64]: Suggestion:

I built a similar logic for the DevOps Self-Hosted construct [here](https://github.com/Azure/DevOps-Self-Hosted/blob/main/constructs/azureImageBuilder/scripts/image/Remove-ImageTemplate.ps1).

There are 2 main differences:
- The script filters for image template not in status `new` or `running`
- I filter all found image template essentially for the prefix. Main reason being - If I run the script multiple times, or something about the pipeline fails or anything, the full image template name will be lost. 
I'd propose to add these 2 features here as well, if you agree they make sense.

File: utilities/pipelines/dependencies/constructs/StoreVhdToStorage/deploymentScripts/Remove-ImageTemplate.ps1 [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+        ImageTemplateName   = $imageTemplateName
+        ResourceGroupName   = $imageTemplateResourceGroup
+    }
+    if ($NoWait) {
+        $resourceActionInputObject['NoWait'] = $true
+    }
+    if ($PSCmdlet.ShouldProcess('Image template [{0}]' -f $imageTemplateName, 'Remove')) {
+        $null = Remove-AzImageBuilderTemplate @resourceActionInputObject
+        Write-Verbose ('Remove image template [{0}] from resource group [{1}]' -f $imageTemplateName, $imageTemplateResourceGroup) -Verbose
```
Comment (MrMCake) [inline utilities/pipelines/dependencies/constructs/StoreVhdToStorage/deploymentScripts/Remove-ImageTemplate.ps1:73]:
```suggestion
        Write-Verbose ('Removed image template [{0}] from resource group [{1}]' -f $imageTemplateName, $imageTemplateResourceGroup) -Verbose
```

File: utilities/pipelines/dependencies/constructs/StoreVhdToStorage/deploymentScripts/Start-ImageTemplate.ps1 [hunk 7]
```diff
@@ -0,0 +1,79 @@
+<#
+    .SYNOPSIS
+    Create artifacts from a given image template
```
Comment (MrMCake) [inline utilities/pipelines/dependencies/constructs/StoreVhdToStorage/deploymentScripts/Start-ImageTemplate.ps1:3]: Why do we call it `artifacts`? If I'm not mistaken, the output will be a new Image Definition `version`. To me, artifacts sounds a bit generic ^^

File: utilities/pipelines/dependencies/constructs/StoreVhdToStorage/deploymentScripts/Start-ImageTemplate.ps1 [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+    $resourceActionInputObject = @{
+        ImageTemplateName   = $imageTemplateName
+        ResourceGroupName   = $imageTemplateResourceGroup
+    }
+    if ($NoWait) {
+        $resourceActionInputObject['NoWait'] = $true
+    }
+    if ($PSCmdlet.ShouldProcess('Image template [{0}]' -f $imageTemplateName, 'Start')) {
+        $null = Start-AzImageBuilderTemplate @resourceActionInputObject
+        Write-Verbose ('Create artifacts from image template [{0}] in resource group [{1}]' -f $imageTemplateName, $imageTemplateResourceGroup) -Verbose
```
Comment (MrMCake) [inline utilities/pipelines/dependencies/constructs/StoreVhdToStorage/deploymentScripts/Start-ImageTemplate.ps1:73]:
```suggestion
        Write-Verbose ('Created/initialized creation of artifacts from image template [{0}] in resource group [{1}]' -f $imageTemplateName, $imageTemplateResourceGroup) -Verbose
```

File: utilities/pipelines/tokensReplacement/Convert-TokensInFileList.ps1 [hunk 9]
```diff
@@ -33,11 +33,13 @@ Optional. A string for a custom output directory of the modified parameter file
...
 
-function Convert-TokensInFile {
+function Convert-TokensInFileList {
     [CmdletBinding()]
     param (
-        [parameter(Mandatory = $true)]
-        [string] $FilePath,
+        [parameter(
+            Mandatory         = $true,
+            ValueFromPipeline = $true)]
```
Comment (MrMCake) [inline utilities/pipelines/tokensReplacement/Convert-TokensInFileList.ps1:41]:
```suggestion
            ValueFromPipeline = $true
            )]
```

File: utilities/tools/Test-NamePrefixAvailability.ps1 [hunk 10]
```diff
@@ -86,7 +86,7 @@ function Test-NamePrefixAvailability {
 
         try {
             # Invoke Token Replacement Functionality and Convert Tokens in Parameter Files
-            $parameterFiles | ForEach-Object { $null = Convert-TokensInFile @ConvertTokensInputs -FilePath $_ }
+            $null = $parameterFiles | Convert-TokensInFileList @ConvertTokensInputs
```
Comment (MrMCake) [inline utilities/tools/Test-NamePrefixAvailability.ps1:89]: This should do the trick now, right?
```suggestion
            $null = Convert-TokensInFileList @ConvertTokensInputs -FilePathList $parameterFiles 
```

File: utilities/tools/Test-NamePrefixAvailability.ps1 [hunk 11]
```diff
@@ -178,7 +178,10 @@ function Test-NamePrefixAvailability {
             # Restore parameter files
             # -----------------------
             Write-Verbose 'Restoring Tokens'
-            $parameterFiles | ForEach-Object { $null = Convert-TokensInFile @ConvertTokensInputs -FilePath $_ -SwapValueWithName $true }
+            $ConvertTokensInputs += @{
+                SwapValueWithName = $true
+            }
+            $null = $parameterFiles | Convert-TokensInFileList @ConvertTokensInputs
```
Comment (MrMCake) [inline utilities/tools/Test-NamePrefixAvailability.ps1:184]: As above:
```suggestion
            $null = Convert-TokensInFileList @ConvertTokensInputs -FilePathList $parameterFiles 
```

File: utilities/tools/Test-ModuleLocally.ps1 [hunk 12]
```diff
@@ -234,7 +234,7 @@ function Test-ModuleLocally {
             }
 
             # Invoke Token Replacement Functionality and Convert Tokens in Parameter Files
-            $moduleTestFiles | ForEach-Object { $null = Convert-TokensInFile @ConvertTokensInputs -FilePath $_ }
+            $moduleTestFiles | Convert-TokensInFileList @ConvertTokensInputs
```
Comment (MrMCake) [inline utilities/tools/Test-ModuleLocally.ps1:237]:
```suggestion
            $null = Convert-TokensInFileList @ConvertTokensInputs -FilePathList $moduleTestFiles
```

File: utilities/tools/Test-ModuleLocally.ps1 [hunk 13]
```diff
@@ -278,7 +278,10 @@ function Test-ModuleLocally {
                 if (($ValidationTest -or $DeploymentTest) -and $ValidateOrDeployParameters) {
                     # Replace Values with Tokens For Repo Updates
                     Write-Verbose 'Restoring Tokens'
-                    $moduleTestFiles | ForEach-Object { $null = Convert-TokensInFile @ConvertTokensInputs -FilePath $_ -SwapValueWithName $true }
+                    $ConvertTokensInputs += @{
+                        SwapValueWithName = $true
+                    }
+                    $null = $moduleTestFiles | Convert-TokensInFileList @ConvertTokensInputs
```
Comment (MrMCake) [inline utilities/tools/Test-ModuleLocally.ps1:284]:
```suggestion
                    $null = Convert-TokensInFileList @ConvertTokensInputs -FilePathList $moduleTestFiles
```

File: utilities/pipelines/resourceRemoval/helper/Get-DependencyResourceNameList.ps1 [hunk 14]
```diff
@@ -49,7 +50,7 @@ function Get-DependencyResourceNameList {
             TokenSuffix = $Settings.parameterFileTokens.tokenSuffix
             Verbose     = $false
         }
-        $parameterFilePaths | ForEach-Object { $null = Convert-TokensInFile @ConvertTokensInputs -FilePath $_ }
+        $null = $parameterFilePaths | Convert-TokensInFileList @ConvertTokensInputs
```
Comment (MrMCake) [inline utilities/pipelines/resourceRemoval/helper/Get-DependencyResourceNameList.ps1:53]:
```suggestion
        $null = Convert-TokensInFileList @ConvertTokensInputs  -FilePathList $parameterFilePaths 
```

File: utilities/pipelines/resourceRemoval/helper/Get-DependencyResourceNameList.ps1 [hunk 15]
```diff
@@ -62,7 +63,10 @@ function Get-DependencyResourceNameList {
 
     if ($Settings.parameterFileTokens.localTokens) {
         Write-Verbose 'Restoring Tokens'
-        $parameterFilePaths | ForEach-Object { $null = Convert-TokensInFile @ConvertTokensInputs -FilePath $_ -SwapValueWithName $true }
+        $ConvertTokensInputs += @{
+            SwapValueWithName = $true
+        }
+        $null = $parameterFilePaths | Convert-TokensInFileList @ConvertTokensInputs
```
Comment (MrMCake) [inline utilities/pipelines/resourceRemoval/helper/Get-DependencyResourceNameList.ps1:69]:
```suggestion
        $null = Convert-TokensInFileList @ConvertTokensInputs -FilePathList $parameterFilePaths 
```

File: .github/actions/templates/validateModuleDeployment/action.yml [hunk 16]
```diff
@@ -279,14 +287,14 @@ runs:
 
     # [Deployment removal] task(s)
     # ----------------------------
-    - name: 'Remove [${{ inputs.templateFilePath }}] from parameters [${{ inputs.parameterFilePath }}]'
+    - name: 'Remove deployed resources'
```
Comment (MrMCake) [inline .github/actions/templates/validateModuleDeployment/action.yml:290]: Ayn reason why we also remove the `templateFilePath` from the output? I was under the impression only the parameter file is the issue.
If that's correct, I'd strongly vote to keep the templateFilePath in as the new dependency approach essentially uses the templateFilePath to differentiate the different test cases :)

File: .github/actions/templates/validateModuleDeployment/action.yml [hunk 17]
```diff
@@ -166,27 +165,33 @@ runs:
...
 
+          # Target files
+          $targetFileList = @('${{ inputs.templateFilePath }}')
+          if(-not [String]::IsNullOrEmpty('${{ inputs.parameterFilePath }}')) {
+            $targetFileList += '${{ inputs.parameterFilePath }}'
+          }
+
           # Invoke Token Replacement Functionality
-          $null = Convert-TokensInFile @ConvertTokensInputs
+          $null = $targetFileList | Convert-TokensInFileList @ConvertTokensInputs
```
Comment (MrMCake) [inline .github/actions/templates/validateModuleDeployment/action.yml:175]:
```suggestion
          $null = Convert-TokensInFileList @ConvertTokensInputs -FilePathList $targetFileList 
```

File: .azuredevops/platformPipelines/platform.dependencies.yml [hunk 18]
```diff
@@ -855,7 +722,7 @@ stages:
                         Write-Verbose ('Using local tokens [{0}]' -f ($tokenMap.Keys -join ', ')) -Verbose
                         $ConvertTokensInputs.Tokens = $tokenMap
                     }
-                    $null = Convert-TokensInFile @ConvertTokensInputs
+                    $null = $parameterFilePaths | Convert-TokensInFileList @ConvertTokensInputs
```
Comment (MrMCake) [inline .azuredevops/platformPipelines/platform.dependencies.yml:725]:
```suggestion
                    $null = Convert-TokensInFileList @ConvertTokensInputs -FilePathList $parameterFilePaths
```

File: .github/workflows/platform.dependencies.yml [hunk 19]
```diff
@@ -373,7 +384,7 @@ jobs:
                 $ConvertTokensInputs.Tokens += $tokenMap
             }
 
-            $null = Convert-TokensInFile @ConvertTokensInputs
+            $null = $parameterFilePaths | Convert-TokensInFileList @ConvertTokensInputs
```
Comment (MrMCake) [inline .github/workflows/platform.dependencies.yml:387]:
```suggestion
            $null = Convert-TokensInFileList @ConvertTokensInputs -FilePathList $parameterFilePaths 
```

File: .github/workflows/platform.dependencies.yml [hunk 20]
```diff
@@ -1028,10 +850,8 @@ jobs:
                 }
                 Write-Verbose ('Using local tokens [{0}]' -f ($tokenMap.Keys -join ', ')) -Verbose
                 $ConvertTokensInputs.Tokens += $tokenMap
-                $NoprConvertTokensInputs.Tokens += $tokenMap
             }
-            $null = Convert-TokensInFile @ConvertTokensInputs
-            $null = Convert-TokensInFile @NoprConvertTokensInputs
+            $null = $parameterFilePaths | Convert-TokensInFileList @ConvertTokensInputs
```
Comment (MrMCake) [inline .github/workflows/platform.dependencies.yml:854]:
```suggestion
            $null = Convert-TokensInFileList @ConvertTokensInputs -FilePathList $parameterFilePaths
```

File: .github/workflows/platform.dependencies.yml [hunk 21]
```diff
@@ -1205,7 +1022,7 @@ jobs:
                 $ConvertTokensInputs.Tokens += $tokenMap
             }
 
-            $null = Convert-TokensInFile @ConvertTokensInputs
+            $null = $parameterFilePaths | Convert-TokensInFileList @ConvertTokensInputs
```
Comment (MrMCake) [inline .github/workflows/platform.dependencies.yml:1025]:
```suggestion
            $null = Convert-TokensInFileList @ConvertTokensInputs -FilePathList $parameterFilePaths 
```

File: .azuredevops/platformPipelines/platform.dependencies.yml [hunk 22]
```diff
@@ -722,10 +592,8 @@ stages:
                     }
                     Write-Verbose ('Using local tokens [{0}]' -f ($tokenMap.Keys -join ', ')) -Verbose
                     $ConvertTokensInputs.Tokens = $tokenMap
-                    $NoprConvertTokensInputs.Tokens = $tokenMap
                 }
-                $null = Convert-TokensInFile @ConvertTokensInputs
-                $null = Convert-TokensInFile @NoprConvertTokensInputs
+                $null = $parameterFilePaths | Convert-TokensInFileList @ConvertTokensInputs
```
Comment (MrMCake) [inline .azuredevops/platformPipelines/platform.dependencies.yml:596]:
```suggestion
                $null = $parameterFilePaths | Convert-TokensInFileList @ConvertTokensInputs -FilePathList $parameterFilePaths 
```

File: .azuredevops/pipelineTemplates/jobs.validateModuleDeployment.yml [hunk 23]
```diff
@@ -169,12 +161,16 @@ jobs:
...
               $ConvertTokensInputs.Tokens += $customTokens
             }
 
+            # Target files
+            $moduleTestFilePath = Join-Path '$(System.DefaultWorkingDirectory)' '$(modulePath)' '$(moduleTestFilePath)'
+            $targetFileList = @($moduleTestFilePath)
+
             # Invoke Token Replacement Functionality
-            $null = Convert-TokensInFile @ConvertTokensInputs
+            $null = $targetFileList | Convert-TokensInFileList @ConvertTokensInputs
```
Comment (MrMCake) [inline .azuredevops/pipelineTemplates/jobs.validateModuleDeployment.yml:173]:
```suggestion
            $null = Convert-TokensInFileList @ConvertTokensInputs -FilePathList $targetFileList 
```

File: .azuredevops/pipelineTemplates/jobs.dependenciesModuleDeployment.yml [hunk 24]
```diff
@@ -145,17 +144,23 @@ jobs:
...
 
+                # Target files
+                $targetFileList = @('${{ deploymentBlock.templateFilePath }}')
+                if(-not [String]::IsNullOrEmpty('${{ deploymentBlock.path }}')) {
+                  $targetFileList += '${{ deploymentBlock.path }}'
+                }
+
                 # Invoke Token Replacement Functionality
-                $null = Convert-TokensInFile @ConvertTokensInputs
+                $null = $targetFileList | Convert-TokensInFileList @ConvertTokensInputs
```
Comment (MrMCake) [inline .azuredevops/pipelineTemplates/jobs.dependenciesModuleDeployment.yml:158]:
```suggestion
                $null = Convert-TokensInFileList @ConvertTokensInputs -FilePathList $targetFileList
```

File: utilities/pipelines/dependencies/constructs/StoreVhdToStorage/deploymentScripts/Remove-ImageTemplate.ps1 (hunk 5 above)
Comment (MrMCake) [reply utilities/pipelines/dependencies/constructs/StoreVhdToStorage/deploymentScripts/Remove-ImageTemplate.ps1:64]: Let's create a dedicated issue for that and add it after this PR is merged

File: utilities/pipelines/dependencies/constructs/StoreVhdToStorage/deploymentScripts/Start-ImageTemplate.ps1 (hunk 7 above)
Comment (MrMCake) [reply utilities/pipelines/dependencies/constructs/StoreVhdToStorage/deploymentScripts/Start-ImageTemplate.ps1:3]: Or call it `image artifacts`

File: .github/actions/templates/validateModuleDeployment/action.yml (hunk 16 above)
Comment (MrMCake) [reply .github/actions/templates/validateModuleDeployment/action.yml:290]: Template file path is still in Job name

File: utilities/pipelines/dependencies/constructs/StoreVhdToStorage/deploymentScripts/Start-ImageTemplate.ps1 (hunk 7 above)
Comment (eriqua) [reply utilities/pipelines/dependencies/constructs/StoreVhdToStorage/deploymentScripts/Start-ImageTemplate.ps1]: Replaced everywhere

File: utilities/pipelines/dependencies/constructs/StoreVhdToStorage/deploymentScripts/Start-ImageTemplate.ps1 (hunk 8 above)
Comment (eriqua) [reply utilities/pipelines/dependencies/constructs/StoreVhdToStorage/deploymentScripts/Start-ImageTemplate.ps1]: Updated as suggested (couldn't apply suggestion)

File: utilities/pipelines/dependencies/constructs/StoreVhdToStorage/deploymentScripts/Copy-VhdToStorageAccount.ps1 (hunk 4 above)
Comment (eriqua) [reply utilities/pipelines/dependencies/constructs/StoreVhdToStorage/deploymentScripts/Copy-VhdToStorageAccount.ps1]: Added

File: utilities/pipelines/dependencies/constructs/StoreVhdToStorage/deploymentScripts/Copy-VhdToStorageAccount.ps1 (hunk 3 above)
Comment (eriqua) [reply utilities/pipelines/dependencies/constructs/StoreVhdToStorage/deploymentScripts/Copy-VhdToStorageAccount.ps1]: Updated

File: utilities/pipelines/tokensReplacement/Convert-TokensInFileList.ps1 (hunk 9 above)
Comment (eriqua) [reply utilities/pipelines/tokensReplacement/Convert-TokensInFileList.ps1]: Updated as suggested (couldn't apply suggestion)

File: utilities/tools/Test-NamePrefixAvailability.ps1 (hunk 10 above)
Comment (eriqua) [reply utilities/tools/Test-NamePrefixAvailability.ps1:89]: I don't have a strong opinion on which of the two use (pipe or explicit parameter). I'll update this and all other occurrences below to use an explicit parameter as suggested, but will add that to the main `@ConvertTokensInputs`

Comment (eriqua) [reply utilities/tools/Test-NamePrefixAvailability.ps1]: done

File: utilities/tools/Test-NamePrefixAvailability.ps1 (hunk 11 above)
Comment (eriqua) [reply utilities/tools/Test-NamePrefixAvailability.ps1]: Same here

File: utilities/tools/Test-ModuleLocally.ps1 (hunk 12 above)
Comment (eriqua) [reply utilities/tools/Test-ModuleLocally.ps1]: Same here

File: utilities/tools/Test-ModuleLocally.ps1 (hunk 13 above)
Comment (eriqua) [reply utilities/tools/Test-ModuleLocally.ps1]: Same here

File: utilities/pipelines/resourceRemoval/helper/Get-DependencyResourceNameList.ps1 (hunk 14 above)
Comment (eriqua) [reply utilities/pipelines/resourceRemoval/helper/Get-DependencyResourceNameList.ps1]: Same here

File: utilities/pipelines/resourceRemoval/helper/Get-DependencyResourceNameList.ps1 (hunk 15 above)
Comment (eriqua) [reply utilities/pipelines/resourceRemoval/helper/Get-DependencyResourceNameList.ps1]: Same here

File: .github/actions/templates/validateModuleDeployment/action.yml (hunk 17 above)
Comment (eriqua) [reply .github/actions/templates/validateModuleDeployment/action.yml]: Same here

File: .azuredevops/platformPipelines/platform.dependencies.yml (hunk 18 above)
Comment (eriqua) [reply .azuredevops/platformPipelines/platform.dependencies.yml]: Same here

File: .github/workflows/platform.dependencies.yml (hunk 19 above)
Comment (eriqua) [reply .github/workflows/platform.dependencies.yml]: Same here

File: .github/workflows/platform.dependencies.yml (hunk 20 above)
Comment (eriqua) [reply .github/workflows/platform.dependencies.yml]: Same here

File: .github/workflows/platform.dependencies.yml (hunk 21 above)
Comment (eriqua) [reply .github/workflows/platform.dependencies.yml]: Same here

File: .azuredevops/platformPipelines/platform.dependencies.yml (hunk 22 above)
Comment (eriqua) [reply .azuredevops/platformPipelines/platform.dependencies.yml]: Same here

File: .azuredevops/pipelineTemplates/jobs.validateModuleDeployment.yml (hunk 23 above)
Comment (eriqua) [reply .azuredevops/pipelineTemplates/jobs.validateModuleDeployment.yml]: Same here

File: .azuredevops/pipelineTemplates/jobs.dependenciesModuleDeployment.yml (hunk 24 above)
Comment (eriqua) [reply .azuredevops/pipelineTemplates/jobs.dependenciesModuleDeployment.yml]: Same here

File: utilities/pipelines/dependencies/constructs/StoreVhdToStorage/deploy.bicep (hunk 2 above)
Comment (eriqua) [reply utilities/pipelines/dependencies/constructs/StoreVhdToStorage/deploy.bicep]: Updated value with the `<<resourceGroupName>>` token

File: utilities/tools/Test-ModuleLocally.ps1 [hunk 25]
```diff
@@ -219,8 +220,8 @@ function Test-ModuleLocally {
             if (Test-Path $settingsFilePath) {
                 $Settings = Get-Content -Path $settingsFilePath -Raw | ConvertFrom-Json -AsHashtable
                 $ConvertTokensInputs += @{
-                    TokenPrefix = $Settings.parameterFileTokens.tokenPrefix
-                    TokenSuffix = $Settings.parameterFileTokens.tokenSuffix
+                    TokenPrefix  = $Settings.parameterFileTokens.tokenPrefix
```
Comment (MrMCake) [inline utilities/tools/Test-ModuleLocally.ps1:223]:
```suggestion
                    TokenPrefix = $Settings.parameterFileTokens.tokenPrefix
```

File: utilities/tools/Test-ModuleLocally.ps1 [hunk 26]
```diff
@@ -219,8 +220,8 @@ function Test-ModuleLocally {
...
+                    TokenSuffix  = $Settings.parameterFileTokens.tokenSuffix
```
Comment (MrMCake) [inline utilities/tools/Test-ModuleLocally.ps1:224]:
```suggestion
                    TokenSuffix = $Settings.parameterFileTokens.tokenSuffix
```

File: .azuredevops/pipelineTemplates/jobs.dependenciesModuleDeployment.yml [hunk 27]
```diff
@@ -110,17 +110,23 @@ jobs:
...
+
                 # Construct Token Function Input
                 $ConvertTokensInputs = @{
-                    Tokens      = @{}
-                    FilePath    = Join-Path '$(System.DefaultWorkingDirectory)' '${{ parameters.modulePath }}' '${{ deploymentBlock.path }}'
-                    TokenPrefix = $Settings.parameterFileTokens.tokenPrefix
-                    TokenSuffix = $Settings.parameterFileTokens.tokenSuffix
+                    FilePathList = $targetFileList
+                    Tokens       = @{}
+                    TokenPrefix  = $Settings.parameterFileTokens.tokenPrefix
```
Comment (MrMCake) [inline .azuredevops/pipelineTemplates/jobs.dependenciesModuleDeployment.yml:128]:
```suggestion
                    TokenPrefix = $Settings.parameterFileTokens.tokenPrefix
```

File: utilities/pipelines/dependencies/constructs/StoreVhdToStorage/deploymentScripts/Remove-ImageTemplate.ps1 (hunk 5 above)
Comment (MrMCake) [reply utilities/pipelines/dependencies/constructs/StoreVhdToStorage/deploymentScripts/Remove-ImageTemplate.ps1:64]: Created issue #1753

**Review by MrMCake — APPROVED** [review]

**Pull request merged**